python/pyarrow/tests/test_table.py

```diff
@@ -493,7 +493,7 @@ def test_recordbatchlist_to_pandas():
 
     table = pa.Table.from_batches([batch1, batch2])
     result = table.to_pandas()
-    data = pd.concat([data1, data2])
+    data = pd.concat([data1, data2]).reset_index(drop=True)
```

Comment:
Why is this now necessary and why did it work before?

Reply:
It worked before because the pandas indexes had been converted to data columns. This change makes the RangeIndex metadata instead